fix: make a valid highlight link required to post a highlight

FatumaA opened the pull request:
## Description
This PR adds a clear error border and message when the highlight is too long. 

Notes: 
I went with adding an extra error span to the existing one so that the too-long error does not block the other errors like the "wrong type of highlight" error when linking an issue.
This is shown in the screenshots below.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Fixes #2560 

## Mobile & Desktop Screenshots/Recordings

![Screenshot 2024-02-05 at 14 34 36](https://github.com/open-sauced/app/assets/67555014/07ef68c6-5b73-48e9-903e-a03799ddfb4e)

![Screenshot 2024-02-05 at 14 34 47](https://github.com/open-sauced/app/assets/67555014/78e8fe05-b0ff-4cb4-9f3a-e2084004de6d)



## Steps to QA
1. Click on the "add a highlight to show your work!" input
2. Paste a description that is longer than 500 characters
3. Notice the only feedback is the char count at the bottom right corner of the text box
4. With this PR, notice the border of the text box turns red and there is a msg displayed at the top of the textbox clearly explaining what the error is


## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?



<!-- note: PRs with deleted sections will be marked invalid -->

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/.github/blob/main/CONTRIBUTING.md.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/.github/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

jpmcb reviewed (approved):
Approved the netlify build. Looks solid 👍🏼

nickytonline commented on components/molecules/HighlightInput/highlight-input-form.tsx:
```diff
@@ -541,12 +541,21 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
                 </span>
               )}
+              {!hasValidCharCount && (
+                <span className="inline-flex items-center gap-2 px-2 py-1 mt-2 text-red-500 bg-red-100 border border-red-500 rounded-md w-full text-sm">
```
@FatumaA, the colors being used cause colour contrast issues.

![CleanShot 2024-02-05 at 07 57 14](https://github.com/open-sauced/app/assets/833231/17a391ea-bee2-4195-844c-825e5edfa245)

![CleanShot 2024-02-05 at 07 58 42](https://github.com/open-sauced/app/assets/833231/f885f0ab-74e3-4b13-ab10-237380e19570)

`bg-red-100` should be white instead.

```suggestion
                <span className="inline-flex items-center gap-2 px-2 py-1 mt-2 text-red-500 bg-white border border-red-500 rounded-md w-full text-sm">
```

takanome-dev reviewed (approved):
Looks great 👍 🍕

FatumaA commented:
> There was another issue that fixed the negative count like you have in your screenshot.
> 
> ![CleanShot 2024-02-08 at 08 20 08@2x](https://private-user-images.githubusercontent.com/833231/303342802-2b33398b-58e4-43f1-ab9b-b6af46e85e5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0MDE2MjEsIm5iZiI6MTcwNzQwMTMyMSwicGF0aCI6Ii84MzMyMzEvMzAzMzQyODAyLTJiMzMzOThiLTU4ZTQtNDNmMS1hYjliLWI2YWY0NmU4NWU1Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwOFQxNDA4NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNTlmOTQxODQwOGU2NzdlMDQ1ZmY2MjJkMDU1MTRlNGYzMWZlZTVmYTA5MWJjY2RjODA1OTBhNmJlY2MwMzc2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OEPSb1OZN0VqZHC-tuKH_H-HUYyHiegH9hQ43AcNBE4)
> 
> Because of that fix, we no longer need to provide a message that your highlight needs to be 500 characters or less because you won't be able to type or past more than 500 characters now.
> 
> See #2613

Ok, so I made the following changes: 
- removed the hasValidCharCount from that file - it will no longer be needed
- removed the error message and error border - they won't be needed

I'm now working around: 
- getting the error messages from the auto-summarize button to clear when the incorrect highlight is replaced with a correct highlight so that the Post button 

I will mark this as draft till I get that behaviour consistent

FatumaA commented:
Hi,
This is ready for review. 
I made this changes:
- removed the hasValidCharCount from that file - it will no longer be needed
- removed the error message and error border - they won't be needed
- got the error messages from the auto-summarize button to clear when the incorrect highlight is replaced with a correct highlight 
- got the add highlight link input to check if link is valid and set error if not
- Post button is disabled when the body text is empty and when the highlight link URL is invalid

Please take a look. 

Thanks!

Video to illustrate changes: 


https://github.com/open-sauced/app/assets/67555014/06550550-3627-4bcb-a072-3a38ce0d6157

brandonroberts commented:
Thanks @FatumaA, one minor change is to clear the error if the field is empty. Everything else looks good

FatumaA commented:
> Thanks @FatumaA, one minor change is to clear the error if the field is empty. Everything else looks good

Hi, 
Done 😌


https://github.com/open-sauced/app/assets/67555014/a08533cf-5c14-4539-ab36-d316d0cdb009

nickytonline reviewed (approved):
:shipit:
